libraries/AP_HAL/utility/functor.h

```diff
@@ -15,10 +15,9 @@
  * You should have received a copy of the GNU General Public License along
  * with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
-
 #pragma once
-#ifndef __FUNCTOR_H__
-#define __FUNCTOR_H__
+
+#include <type_traits>
```

Comment:
this breaks the qurt build, with:
/home/tridge/project/UAV/APM/libraries/AP_HAL/utility/functor.h:20:10: fatal error: 'type_traits' file not found
looks like we don't get type_trains on clang with -std=gnu++11